tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -467,14 +460,71 @@ export class WebGPUBackend extends KernelBackend {
       vals = backend_util.mergeRealAndImagArrays(
           realValues as Float32Array, imagValues as Float32Array);
     } else {
-      const data = await this.getBufferData(info);
+      const data = info.values != null ?
+          info.values :
+          await this.getBufferData(
+              info.bufferInfo.buffer, info.bufferInfo.byteSize);
       vals =
           webgpu_util.ArrayBufferToTypedArray(data as ArrayBuffer, info.dtype);
     }
     this.convertAndCacheOnCPU(dataId, vals);
     return vals;
   }
 
+  /**
+   * Read tensor to a new GPUBuffer.
+   * @param dataId The source tensor.
+   * @param options
+   *     customBufShape: Optional. If set, will use the user defined buffer
+   *     size to create the buffer.
+   */
+  readToGPU(dataId: DataId, options: DataToGPUWebGLOption = {}): GPUData {
+    const srcData = this.tensorMap.get(dataId);
+    const {values, dtype, bufferInfo} = srcData;
+
+    if (dtype === 'complex64') {
+      throw new Error('Does not support reading buffer for complex64 dtype.');
+    }
+
+    if (bufferInfo.buffer == null) {
+      if (values != null) {
+        throw new Error('Data is not on GPU but on CPU.');
+      } else {
+        throw new Error('There is no data on GPU or CPU.');
+      }
+    }
+
+    if (options.customBufSize != null) {
+      util.assert(
+          options.customBufSize >= bufferInfo.byteSize,
+          () => 'customBufSize should be equal or larger than the buffer size');
+    }
+
+    const bufferSize = options.customBufSize != null ?
+        Math.max(bufferInfo.byteSize, options.customBufSize) :
+        bufferInfo.byteSize;
+    const copySize = options.customBufSize != null ?
```

Comment:
`const copySize = bufferInfo.byteSize;`? Or directly use `bufferInfo.byteSize` in `copyBufferToBuffer`.

Reply:
Done